esp32-mock-generator/src/main.cpp

```
@@ -0,0 +1,319 @@
+                           - powerActive * powerActive);
+
+  // -------------------------------------------------------------------------
+  // 3. Generate Harmonics
+  // -------------------------------------------------------------------------
+  bool highTHD = (randomFloat(0, 1) < PROBABILITY_HIGH_THD);
+  if (highTHD && !eventOccurred) {
+    Serial.print("    ⚠️  HIGH THD ");
+    eventOccurred = true;
+  }
+
+  float harmonicsV[8];  // Voltage harmonics (H1 to H8)
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)